server/build/Dockerfile

```diff
@@ -34,15 +31,41 @@ RUN mkdir -p /mattermost/data /mattermost/plugins /mattermost/client/plugins \
   && curl -L $MM_PACKAGE | tar -xvz \
   && chown -R mattermost:mattermost /mattermost /mattermost/data /mattermost/plugins /mattermost/client/plugins
 
+# Final stage using distroless for minimal attack surface
+FROM gcr.io/distroless/base-debian12
+
+# Some ENV variables
+ENV PATH="/mattermost/bin:${PATH}"
+
+# Copy document processing utilities and necessary support files
+COPY --from=builder /usr/bin/pdftotext /usr/bin/pdftotext
+COPY --from=builder /usr/bin/wvText /usr/bin/wvText
+COPY --from=builder /usr/bin/wvWare /usr/bin/wvWare
+COPY --from=builder /usr/bin/unrtf /usr/bin/unrtf
+COPY --from=builder /usr/bin/tidy /usr/bin/tidy
+COPY --from=builder /usr/share/wv /usr/share/wv
+
+# Copy necessary libraries for document processing utilities
+COPY --from=builder /usr/lib/libpoppler.so* /usr/lib/
+COPY --from=builder /usr/lib/libfreetype.so* /usr/lib/
+COPY --from=builder /usr/lib/libpng.so* /usr/lib/
+COPY --from=builder /usr/lib/libwv.so* /usr/lib/
+COPY --from=builder /usr/lib/libtidy.so* /usr/lib/
+COPY --from=builder /usr/lib/libfontconfig.so* /usr/lib/
+
+# Copy mattermost from builder stage 
+COPY --from=builder --chown=2000:2000 /mattermost /mattermost
+
 # We should refrain from running as privileged user
-USER mattermost
+USER 2000:2000
```

Comment:
Should we adopt `65532`/`nonroot` to be in-line with what Distroless expect, or do we need to create the mattermost user manually in `etc/passwd`?

I assume the latter is easier as it'd be a non-breaking change, but not sure if Distroless supports it. The former may break existing Docker deployments.

Reply:
Backward compatibility is what I had in mind. If I'm reading the section "Specify user by UID and GID" right: https://www.docker.com/blog/understanding-the-docker-user-instruction we don't need to create the mattermost user manually in etc/passwrd

Reply:
You're right. We needed to add the mattermost user in `/etc/passwd` and that worked.